### PR TITLE
#O6QG2WdZ Reset the password

### PR DESCRIPTION
**Pull request by Ally4**
#### What does this PR do?

- This PR reset the password of a user who is in the system

#### Description of Task to be completed?

- create a controller for reset link

- create a validation for reset link

- create a router for reset link

#### How should this be manually tested?

- clone this repo

- run `npm i`

- Create a .env file and fill it with your own credentials taken example as in the .env.example 

- Create a SendGrid account and its key in the .env

- run `npm run migrate:reset`

- run `npm run seed`

- run `npm run dev`

- Create a SendGrid account and its key in the .env

- Test the endpoints in this documentation: `http://localhost:3020/api-docs/#/`

- After getting the link in the email, take the token and put it in the parameter of the reset-password,  and it starts right after `reset-password/`

- Make sure you put the token from admin login when token is asked in the forgot-link documentation

#### What are the relevant pivotal tracker stories?

- https://trello.com/c/O6QG2WdZ/22-reset-password

#### The screenshots related?

Screen Shot 2020-10-27 at 06 59 03
Screen Shot 2020-10-27 at 06 59 12
Screen Shot 2020-10-27 at 06 59 25
Screen Shot 2020-10-27 at 06 59 40

### CODE REVIEW

File: src/controllers/userController.js [hunk 1]
```diff
@@ -23,13 +24,14 @@ class register {
         role,
       } = req.body;
       const id = uuidv4();
+      console.log(`this is the email ------------------${email}----------------------`);
```
Comment (Ntwali-Josue) [inline src/controllers/userController.js]: can you remove this console.log line

Comment (Ally4) [reply src/controllers/userController.js]: I removed it

**Review by Ntwali-Josue — APPROVED** [review]

**Review by Mugema123 — APPROVED** [review]

**Review by TuyizeeAnastase — APPROVED** [review]

**Review by blaise82 — COMMENTED** [review]
- remove the token field from your swagger documentation

File: package.json [hunk 2]
```diff
@@ -10,21 +10,14 @@
     "dev": "cross-env NODE_ENV=development nodemon --exec babel-node  ./src/app.js",
     "migrate:reset": "npx sequelize-cli db:migrate:undo:all && npm run migrate",
     "migrate": "sequelize db:migrate",
-    "seed": "sequelize db:seed:all",
+    "seed": "npx sequelize-cli db:seed:all",
```
Comment (blaise82) [inline package.json]: remove npx

**Review by blaise82 — CHANGES_REQUESTED** [review]
- make sure  that you have users table

File: src/middleware/isDriverOrOperator.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  const { email } = req.body;
+  const user = await Users.findOne({ where: { email } });
+  if (!user) {
+    res.status(403).json({
+      status: 403,
+      message: res.__('User with that email is not found'),
+    });
+    return false;
+  }
+  if (user.role !== 'operator' && user.role !== 'driver') {
```
Comment (blaise82) [inline src/middleware/isDriverOrOperator.js:15]: why are you using and?

**Review by blaise82 — CHANGES_REQUESTED** [review]

File: src/middleware/isDriverOrOperator.js (hunk 3 above)
Comment (Ally4) [reply src/middleware/isDriverOrOperator.js:15]: Because there is NOT EQUAL (!==)

File: package.json (hunk 2 above)
Comment (Ally4) [reply package.json]: done

**Review by blaise82 — APPROVED** [review]

**Pull request merged**